api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -553,18 +553,29 @@ public void setDataUseAgreementNameOutOfDate(String newGivenName, String newFami
     userDataUseAgreementDao.saveAll(dataUseAgreements);
   }
 
-  @Override
-  @Transactional
-  public void submitTermsOfService(DbUser dbUser, Integer tosVersion) {
+  /**
+   * Validates a given tosVersion, by running all validation checks.
+   * 
+   * @param tosVersion
+   * @throws BadRequestException
```

Comment:
personally I don't use @throws for unchecked exception. But that's up to you. If using, I think it usually followed by `if .....`
See go/java-practices/javadoc#throws

Reply:
sorry, I don't get what do you mean by "if ....." ?

Reply:
See the last sentence in go/java-practices/javadoc#throws
`When @throws SomeException is used, it is almost always followed by the word if and an explanation of the condition in which that exception is thrown`